src/App.tsx

```diff
@@ -126,11 +127,30 @@ const router = createBrowserRouter([
 
 export default function App() {
   const {darkMode} = useSettings();
+  const logError = useLogError();
 
   useEffect(() => {
     document.documentElement.classList.toggle('dark', darkMode);
   }, [darkMode]);
 
+  useEffect(() => {
+    function onError(event: ErrorEvent) {
+      logError(event.error);
+    }
+
+    function onUnhandledRejection(event: PromiseRejectionEvent) {
+      logError(event.reason);
+    }
+
+    window.addEventListener('error', onError);
+    window.addEventListener('unhandledrejection', onUnhandledRejection);
+
+    return () => {
+      window.removeEventListener('error', onError);
+      window.removeEventListener('unhandledrejection', onUnhandledRejection);
+    };
+  });
```

Comment:
wouldn't it make more sense to move that inside `useLogError`?

Reply:
This hook could in principle be used in multiple components in which case we'd end up registering the error handlers more than once. So I think it might be better to leave it here